src/MatrixClientPeg.js

```diff
@@ -35,73 +32,22 @@ function deviceId() {
     return id;
 }
 
-function createClientForPeg(hs_url, is_url, user_id, access_token, guestAccess) {
-    var opts = {
-        baseUrl: hs_url,
-        idBaseUrl: is_url,
-        accessToken: access_token,
-        userId: user_id,
-        timelineSupport: true,
-    };
-
-    if (localStorage) {
-        opts.sessionStore = new Matrix.WebStorageSessionStore(localStorage);
-        opts.deviceId = deviceId();
-    }
-
-    matrixClient = Matrix.createClient(opts);
-
-    // we're going to add eventlisteners for each matrix event tile, so the
-    // potential number of event listeners is quite high.
-    matrixClient.setMaxListeners(500);
-
-    if (guestAccess) {
-        console.log("Guest: %s", guestAccess.isGuest());
-        matrixClient.setGuest(guestAccess.isGuest());
-        var peekedRoomId = guestAccess.getPeekedRoom();
-        if (peekedRoomId) {
-            console.log("Peeking in room %s", peekedRoomId);
-            matrixClient.peekInRoom(peekedRoomId);
-        }
-    }
-}
-
-if (localStorage) {
-    var hs_url = localStorage.getItem("mx_hs_url");
-    var is_url = localStorage.getItem("mx_is_url") || 'https://matrix.org';
-    var access_token = localStorage.getItem("mx_access_token");
-    var user_id = localStorage.getItem("mx_user_id");
-    var guestAccess = new GuestAccess(localStorage);
-    if (access_token && user_id && hs_url) {
-        console.log("Restoring session for %s", user_id);
-        createClientForPeg(hs_url, is_url, user_id, access_token, guestAccess);
-    }
-    else {
-        console.log("Session not found.");
-    }
-}
-
-class MatrixClient {
-
+// A thing that holds your Matrix Client
+// Also magically works across sessions through the power of localstorage
+class MatrixClientPeg {
     constructor(guestAccess) {
+        this.matrixClient = null;
         this.guestAccess = guestAccess;
     }
 
     get(): MatrixClient {
-        return matrixClient;
+        return this.matrixClient;
     }
 
     unset() {
-        matrixClient = null;
+        this.matrixClient = null;
     }
 
-    // FIXME, XXX: this all seems very convoluted :(
-    //
-    // Why do we have this peg wrapper rather than just MatrixClient.get()?
```

Comment:
I'd hoped I cleared this up - I've renamed everything in here to be MatrixClientPeg consistently, avoiding the name MatrixClient because it's not actually the client object returned by the js-sdk.